### ISSUE TITLE
configure beats

### ISSUE THREAD

**Issue by danlaunl** [MartinSStewart/elm-audio#13]
I'd like to make an interface to specify the beats per minute (ie # loops per minute) with which play a sample(s) of arbitrary length.

Either a) it should loop (this library's LoopConfig)  _with silence_ for a  duration , calculated from Audio.length . (utilising playbackrate is insufficient bc it changes pitch)

Or b)  avoid the LoopConfig and write a different data structure to load the same file with  a sequence of start times ? (for tens of samples being beated out in parallel, and ad infinitum)  (still performant?)

I don't know any js so am trying to avoid touching audio.js

have already enjoyed connecting the library to a playful interface messing around with multiple sources and volumes!

**Comment by MartinSStewart** [comment]
My suggestion was going to be something like this
```elm
let
    default =
        Audio.audioDefaultConfig
in
-- Play the drumSound at 15 beats per minute
Audio.audioWithConfig
    { default | loop = Just { loopStart = Duration.seconds 0, loopEnd = Duration.seconds 4 } }
    drumSound
    time
```
but when I tried it, I noticed that if the sound effect is shorter than the loop length, it doesn't wait the full loop duration before playing again. This is a bug I'll need to fix.

In the meantime, my suggestion is to make sure your samples have a lot of silence at the end of them so the sample is longer than the loop duration.

I have some free time today so hopefully I can fix this quickly.
